### ISSUE TITLE
How do I avoid overfitting when training Flair for new NER task

### ISSUE THREAD

**Issue by willer2009** [flairNLP/flair#1911]
Hello I'm try to use flair for the task of recognizing skills from Job advertisements and module description in university syllabus.
This is done for the German language.
I've manually (and painfully) selected ca. 3000 skills from job descriptions that I use to annotate sentences from  job advertisements. My intuition/plan was to use the trained model to discover new skills from job ads and incrementally have more annotated distinct data with time.
At the moment the trained model works almost like a dictionary identifying mostly data which are present in my list of 3000 skills.
I think this is a classical case of overfitting, right? How can I prevent this and make flair learn more about the context where a skill appears.
One thing is also that the data are not normalized at the moment: I have almost 90000 annotated sentences for only 3000 (almost) distinct skills, with some skill appearing a lot more often that others in the annotated corpus.
Here is what my training code looks like:

**# define columns**
columns = {0: 'text', 1: 'ner'}

**# this is the folder in which train, test and dev files reside**
data_folder = 'training_data'

**# init a corpus using column format, data folder and the names of the train, dev and test files**
corpus: Corpus = ColumnCorpus(data_folder, columns,
                              train_file='train_full.txt',
                              test_file='test_full.txt',
                              dev_file='dev_full.txt')

len(corpus.train)
print(corpus.train[0].to_tagged_string('ner'))

**# 2. what tag do we want to predict?**
tag_type = 'ner'

**# 3. make the tag dictionary from the corpus**
tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
#print(tag_dictionary)

**# 4. initialize embeddings**
embedding_types: List[TokenEmbeddings] = [

    WordEmbeddings('de'),

    # comment in this line to use character embeddings
    #CharacterEmbeddings(),

    # comment in these lines to use flair embeddings
    FlairEmbeddings('de-forward'),
    FlairEmbeddings('de-backward'),
]

embeddings: StackedEmbeddings = StackedEmbeddings(embeddings=embedding_types)

**# 5. initialize sequence tagger**
from flair.models import SequenceTagger

tagger: SequenceTagger = SequenceTagger(hidden_size=256,
                                        embeddings=embeddings,
                                        tag_dictionary=tag_dictionary,
                                        tag_type=tag_type,
                                        use_crf=True)

**# 6. initialize trainer**
from flair.trainers import ModelTrainer

trainer: ModelTrainer = ModelTrainer(tagger, corpus)

**# 7. start training**
trainer.train('training_result_full_forward',
              learning_rate=0.1,
              mini_batch_size=32,
              max_epochs=150,
              embeddings_storage_mode='none',
              checkpoint=True)

my questions now:
1- Should I use only each skill once in the training data, therefore the test and dev set would have different skills and this will force Flair to improve in context learning?
2- Is it possible to associate POS information to the data during NER training (using a format like columns = {0: 'text', 1:'pos', 2: 'ner'}), or is flair doing this anyway under the wood? One thing I taught of was to associate the text with pos and have something like text-pos (e.g- Python-PROPN instead of just Python as text in the first column to force FLAIR to consider the POS information)

**Comment by Swayatta** [comment]
Were you able to get a solution to this ?
